src/CostFunction3D.cpp

```diff
@@ -2558,32 +2567,25 @@ void CostFunction3D::calcHmatrix()
               if (!kbasis) continue;
               // Count the number of non-zero entries in the observation matrix...
               hi++;
+              nnz++;
             }
           }
         }
       }
     }
-    Hlength[m]=hi;
   }
 
-  integer nonzeros = 0;
-  for (int m = 0; m < mObs; m++) {
-    nonzeros += Hlength[m];
-  }
-  //GPTLstop("CostFunction3D::calcHmatrix:nonzeros");
-
-  IH[mObs] = nonzeros;
-  std::cout << "sizeof(integer): " << sizeof(integer) << "\n";
-  std::cout << "Non-zero entries in sparse H matrix: " << nonzeros << " = " << 100.0*float(nonzeros)/(float(mObs)*float(nState)) << " %\n";
-  std::cout << "Memory usage for [H]             (Mbytes): " << sizeof(real)*(nonzeros)/(1024.0*1024.0) << "\n";
-  H    = new real[nonzeros];
-  JH   = new integer [nonzeros];
-  mVal = new integer [nonzeros];
-  mTmp = new integer [nonzeros];
-  mIncr = new integer [nState];
+  //GPTLstop("CostFunction3D::calcHmatrix:nnz");
+  IH[mObs] = nnz;
 
+  std::cout << "CostFunction3D:: nnz " << nnz << std::endl;
```

Comment:
I guess `nnz` here refers to the non-zero entries in the sparse H matrix? If so, I prefer the original message that gives more details and is straightforward.

Reply:
I reverted it back to include nonzeros